DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,37 @@
+# Developer Guide
+
+## Setup
+
+```sh
+make toolchain # Install required to develop the project
+```
+
+## Testing a code change
+
+Deploy your changes to a local development cluster and run the tests against it.
+
+```sh
+make apply # Apply your changes
+make e2etest # Run the integration test suite
+```
+
+In another terminal, you can tail the logs with stern
+```sh
+stern -l app=vpc-resource-controller -n kube-system
+```
+
+## Submitting a PR
+Run the presubmit target and check in all generated code before submitting a PR.
+
+```sh
+make presubmit
+```
+
+## Troubleshooting
+
+### Invalid value 'trunk' for InterfaceType
+
+The following error means that must be allowlisted for EC2 Networking
```

Comment:
what must be allow-listed?

Reply:
The trunk API need approval on AWS account level.